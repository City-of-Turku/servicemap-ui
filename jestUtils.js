import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });