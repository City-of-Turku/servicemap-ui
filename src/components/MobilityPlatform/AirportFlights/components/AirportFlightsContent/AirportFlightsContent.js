import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { Typography } from '@mui/material';
import { css } from '@emotion/css';
import { format } from 'date-fns';
import { ReactSVG } from 'react-svg';
import styled from '@emotion/styled';
import planeIcon from 'servicemap-ui-turku/assets/icons/icons-icon_airplane-square.svg';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../styled/styled';

const AirportFlightsContent = ({ arrivals, departees }) => {
  const intl = useIntl();

  const iconClass = css({
    fill: 'rgba(7, 44, 115, 255)',
    width: '18px',
    height: '18px',
  });

  const currentDate = new Date();
  const formatDate = dateValue => format(new Date(dateValue), 'MM-dd');
  const formatDateTime = dateTimeValue => format(new Date(dateTimeValue), 'dd.MM HH:mm');

  const filterArrivals = arrivals.filter(item => item.sdt.includes(formatDate(currentDate)));
  const filterDepartees = departees.filter(item => item.sdt.includes(formatDate(currentDate)));

  const getLocalizedCities = cityStr => {
    switch (cityStr) {
      case 'Stockholm':
        return intl.formatMessage({ id: 'mobilityPlatform.content.airport.cities.stockholm' });
      case 'Mariehamn':
        return intl.formatMessage({ id: 'mobilityPlatform.content.airport.cities.mariehamn' });
      case 'Riga':
        return intl.formatMessage({ id: 'mobilityPlatform.content.airport.cities.riga' });
      case 'Gdansk':
        return intl.formatMessage({ id: 'mobilityPlatform.content.airport.cities.gdansk' });
      default:
        return cityStr;
    }
  };

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        <Typography variant="subtitle1" component="h3">
          {intl.formatMessage({ id: 'mobilityPlatform.content.airport.title' })}
        </Typography>
      </StyledHeaderContainer>
      <div>
        <StyledTextContainer>
          <Typography variant="subtitle1" component="p">
            {intl.formatMessage({ id: 'mobilityPlatform.content.airport.departees' })}
          </Typography>
        </StyledTextContainer>
        {filterDepartees.map(item => (
          <StyledFlexContainer key={item.sdt}>
            <StyledTextContainer>
              <ReactSVG src={planeIcon} className={iconClass} />
            </StyledTextContainer>
            <StyledTextContainer>
              <Typography variant="body2" component="p">
                {item.fltnr}
              </Typography>
            </StyledTextContainer>
            <StyledTextContainer>
              <Typography variant="body2" component="p">
                {getLocalizedCities(item.route_n_1)}
              </Typography>
            </StyledTextContainer>
            <StyledTextContainer>
              <Typography variant="body2" component="p">
                {formatDateTime(item.sdt)}
              </Typography>
            </StyledTextContainer>
          </StyledFlexContainer>
        ))}
      </div>
      <div>
        <StyledTextContainer>
          <Typography variant="subtitle1" component="p">
            {intl.formatMessage({ id: 'mobilityPlatform.content.airport.arrivals' })}
          </Typography>
        </StyledTextContainer>
        {filterArrivals.map(item => (
          <StyledFlexContainer key={item.sdt}>
            <StyledTextContainer>
              <ReactSVG src={planeIcon} className={iconClass} />
            </StyledTextContainer>
            <StyledTextContainer>
              <Typography variant="body2" component="p">
                {item.fltnr}
              </Typography>
            </StyledTextContainer>
            <StyledTextContainer>
              <Typography variant="body2" component="p">
                {intl.formatMessage({ id: 'mobilityPlatform.content.airport.cities.turku' })}
              </Typography>
            </StyledTextContainer>
            <StyledTextContainer>
              <Typography variant="body2" component="p">
                {formatDateTime(item.sdt)}
              </Typography>
            </StyledTextContainer>
          </StyledFlexContainer>
        ))}
      </div>
    </StyledContainer>
  );
};

const StyledFlexContainer = styled.div(({ theme }) => ({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
  marginTop: theme.spacing(0.5),
  marginBottom: theme.spacing(0.75),
  width: '93%',
}));

AirportFlightsContent.propTypes = {
  arrivals: PropTypes.arrayOf(PropTypes.shape({
    sdt: PropTypes.string,
    fltnr: PropTypes.string,
    route_n_1: PropTypes.string,
  })),
  departees: PropTypes.arrayOf(PropTypes.shape({
    sdt: PropTypes.string,
    fltnr: PropTypes.string,
    route_n_1: PropTypes.string,
  })),
};

AirportFlightsContent.defaultProps = {
  arrivals: [],
  departees: [],
};

export default AirportFlightsContent;
