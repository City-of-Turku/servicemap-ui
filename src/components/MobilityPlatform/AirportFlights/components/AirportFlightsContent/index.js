import AirportFlightsContent from './AirportFlightsContent';

export default AirportFlightsContent;
