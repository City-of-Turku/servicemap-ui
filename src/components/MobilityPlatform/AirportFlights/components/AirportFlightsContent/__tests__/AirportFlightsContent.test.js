// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../i18n/fi';
import AirportFlightsContent from '../index';

const mockProps = {
  departees: [
    {
      sdt: '2024-04-26T11:00:00Z',
      fltnr: 'TST1',
      route_n_1: 'Stockholm',
    },
    {
      sdt: '2024-04-26T12:00:00Z',
      fltnr: 'TST2',
      route_n_1: 'Riga',
    },
    {
      sdt: '2024-04-26T17:00:00Z',
      fltnr: 'TST3',
      route_n_1: 'Rome',
    },
  ],
  arrvals: [
    {
      sdt: '2024-04-26T10:00:00Z',
      fltnr: 'TST1',
      route_n_1: 'Turku',
    },
    {
      sdt: '2024-04-26T15:00:00Z',
      fltnr: 'TST2',
      route_n_1: 'Turku',
    },
    {
      sdt: '2024-04-26T20:00:00Z',
      fltnr: 'TST3',
      route_n_1: 'Turku',
    },
  ],
};

const renderWithProviders = getRenderWithProviders({});

describe('<AirportFlightsContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<AirportFlightsContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text', () => {
    const { container } = renderWithProviders(<AirportFlightsContent {...mockProps} />);
    const p = container.querySelectorAll('p');
    const h4 = container.querySelector('h4');
    expect(h4.textContent).toContain(finnishTranslations['mobilityPlatform.content.airport.title']);
    expect(p[0].textContent).toContain(finnishTranslations['mobilityPlatform.content.airport.departees']);
    expect(p[1]).toBeInTheDocument();
  });
});
