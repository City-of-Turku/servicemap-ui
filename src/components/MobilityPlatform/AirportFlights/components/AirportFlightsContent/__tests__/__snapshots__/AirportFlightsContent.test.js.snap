// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AirportFlightsContent /> should match snapshot 1`] = `
<div>
  <div
    class="css-1ynyhby"
  >
    <div
      class="css-1dznbe2"
    >
      <h4
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Turun lentoasema
      </h4>
    </div>
    <div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          L채htev채t lennot:
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Ei l채htevi채 lentoja.
        </p>
      </div>
    </div>
    <div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          Saapuvat lennot:
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Ei saapuvia lentoja.
        </p>
      </div>
    </div>
  </div>
</div>
`;
